source/dofs/dof_tools_constraints.cc

```diff
@@ -1070,6 +1101,1316 @@ namespace DoFTools
+                                            fe.constraints()(row + shift_1,
+                                                             dof_i_on_mother));
+                    }
+                }
+            }
+        }
+    }
+
+
+    template <int dim_, int spacedim, typename number>
+    void
+    make_hanging_node_constraints_nedelec(
+      const DoFHandler<dim_, spacedim> &dof_handler,
+      AffineConstraints<number>        &constraints,
+      std::integral_constant<int, 3>)
+    {
+      const unsigned int dim = 3;
+
+      std::vector<types::global_dof_index> dofs_on_mother;
+      std::vector<types::global_dof_index> dofs_on_children;
+
+      // loop over all quads; only on quads there can be constraints. We do so
+      // by looping over all active cells and checking whether any of the faces
+      // are refined which can only be from the neighboring cell because this
+      // one is active. In that case, the face is subject to constraints
+      //
+      // note that even though we may visit a face twice if the neighboring
+      // cells are equally refined, we can only visit each face with hanging
+      // nodes once
+      for (const auto &cell : dof_handler.active_cell_iterators())
+        {
+          // Skip artificial cells:
+          // artificial cells can at best neighbor ghost cells, but we're not
+          // interested in these interfaces
+          if (cell->is_artificial())
+            continue;
+
+          // loop over all faces
+          for (const unsigned int face : cell->face_indices())
+            {
+              // skip cells without children
+              if (cell->face(face)->has_children() == false)
+                continue;
+
+              // first of all, make sure that we treat a case which is
+              // possible, i.e. either no dofs on the face at all or no
+              // anisotropic refinement
+              if (cell->get_fe().n_dofs_per_face(face) == 0)
+                continue;
+
+              Assert(cell->face(face)->refinement_case() ==
+                       RefinementCase<dim - 1>::isotropic_refinement,
+                     ExcNotImplemented());
+
+              // in any case, faces can have at most two active FE indices,
+              // but here the face can have only one (namely the same as that
+              // from the cell we're sitting on), and each of the children can
+              // have only one as well. check this
+              AssertDimension(cell->face(face)->n_active_fe_indices(), 1);
+
+              Assert(cell->face(face)->fe_index_is_active(
+                       cell->active_fe_index()) == true,
+                     ExcInternalError());
+
+              for (unsigned int c = 0; c < cell->face(face)->n_children(); ++c)
+                if (!cell->neighbor_child_on_subface(face, c)->is_artificial())
+                  AssertDimension(
+                    cell->face(face)->child(c)->n_active_fe_indices(), 1);
```

Comment:
Maybe, just combine with the loop below?

Reply:
Similar to above, I combined the loops and added #ifdef DEBUG (...) #endif.